chain/assembler.go

```diff
@@ -5,70 +5,46 @@ package chain
 
 import (
 	"context"
-	"time"
 
-	"github.com/ava-labs/hypersdk/utils"
+	"github.com/ava-labs/hypersdk/state"
 )
 
-type Assembler struct {
-	vm VM
-}
-
-func (a *Assembler) AssembleBlock(
+func (c *Chain) AssembleBlock(
 	ctx context.Context,
-	parent *StatefulBlock,
+	parentView state.View,
+	parent *ExecutionBlock,
```

Comment:
Q: Why we can't just use the `StatelessBlock` type here? Not really critical that we address anything in this file now since this is dead code anyways though

Reply:
We could use `StatelessBlock` in DSMR since execution occurs only accept, so there's no need to kick off async verification separately on parse.

We could kick off async signature verification of transactions received in chunks instead